### PR TITLE
Internal cleanups

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: libnopegl/src/pgcraft.c [hunk 1]
```diff
@@ -1271,6 +1271,33 @@ static void probe_texture_infos(struct pgcraft *s)
     }
 }
 
+static void probe_ublocks(struct pgcraft *s)
+{
+    struct darray *array = &s->filtered_pipeline_info.desc.buffers;
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:1276]: `const`?

File: libnopegl/src/pgcraft.c [hunk 2]
```diff
@@ -1271,6 +1271,33 @@ static void probe_texture_infos(struct pgcraft *s)
...
+    const struct pipeline_resource_desc *buffer_descs = ngli_darray_data(array);
+    const size_t nb_buffer_descs = ngli_darray_count(array);
+
+    struct pgcraft_compat_info *info = &s->compat_info;
+    for (size_t i = 0; i < NGLI_PROGRAM_SHADER_NB; i++) {
+        struct block *block = &info->ublocks[i];
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:1282]: `const`?

File: libnopegl/src/pgcraft.c [hunk 3]
```diff
@@ -1271,6 +1271,33 @@ static void probe_texture_infos(struct pgcraft *s)
     }
 }
 
+static void probe_ublocks(struct pgcraft *s)
+{
+    struct darray *array = &s->filtered_pipeline_info.desc.buffers;
+    const struct pipeline_resource_desc *buffer_descs = ngli_darray_data(array);
+    const size_t nb_buffer_descs = ngli_darray_count(array);
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:1278]: can be in the loop; we do that everywhere

File: libnopegl/src/pgcraft.c (hunk 1 above)
Comment (mbouron) [reply libnopegl/src/pgcraft.c]: Done.

File: libnopegl/src/pgcraft.c (hunk 3 above)
Comment (mbouron) [reply libnopegl/src/pgcraft.c]: Done.

Comment (mbouron) [reply libnopegl/src/pgcraft.c]: And move `const struct pipeline_resource_desc *buffer_descs = ngli_darray_data(array);` just before the loop

**Review by ubitux — APPROVED** [review]

**Pull request merged**